preprocessing/data_explorer.py

```diff
@@ -147,3 +149,105 @@ def explore_data(dataframe):
         explore_object.data_explore,
         column_i=widgets.IntSlider(min=0, max=dataframe.shape[1] - 1, step=1, value=0),
     )
+
+
+class OutliersDetection:
+
+    def __init__(self, data_array: np.array, number_trees: int = 100, tree_size: int = 128):
+        """ Parameters initiator
+
+        :param np.array data_array: A numpy array which has the size of n x m. The missing values are replaced by the
+                                    means and the elements are only numeric
+        :param int number_trees: The number of the trees that will be randomly generated to create the forest
+        :param int tree_size: The number of the elements that a tree should contain.
+        """
+
+        self.data_array = data_array
+        self.number_trees = number_trees
+        self.tree_size = tree_size
+        self.number_of_rows = self.data_array.shape[0]
+        self.number_of_feature = self.data_array.shape[1]
+
+    def construct_force(self) -> List:
+        """ Forest creator
+
+        This function creates a list of trees which are constructed randomly.
+
+        :return:
+                - list forest: A list of the trees that are created randomly
+        """
+
+        forest = []
+
+        sample_size_range = (self.number_of_rows // self.tree_size, self.tree_size)
+
+        if sample_size_range[0] == 0:
+            raise ValueError("Please check the tree_size. It seems that the number of the "
+                             "samples (rows) is less than the size of the tree")
+
+        while len(forest) < self.number_trees:
+            # Select random subsets of points uniformly from point set
+            ixs = np.random.choice(self.number_of_rows, size=sample_size_range,
+                                   replace=False)
+            # Add sampled trees to forest
+            trees = [rrcf.RCTree(self.data_array[ix], index_labels=ix)
+                     for ix in ixs]
+            forest.extend(trees)
+        return forest
+
+    def codisp_average(self, forest) -> pd.Series:
```

Comment:
`forest: list`
?

Reply:
I wish you gave more content about this comment so that I would understand it clearly.  I think you ask me to add the list in front of the variable forset.